logger.h

```diff
@@ -17,22 +17,82 @@
 #define TO_S(M) #M
 #define STR(M) TO_S(M)
 
+#ifdef _LOG_SYSLOG
+#undef _LOG_SYSLOG
+#define _LOG_SYSLOG 1
+#else
+#define _LOG_SYSLOG 0
+#endif // _LOG_SYSLOG
+
+#ifdef _LOG_STDOUT
+#undef _LOG_STDOUT
+#define _LOG_STDOUT 1
+#else
+#define _LOG_STDOUT 0
+#endif // _LOG_STDOUT
+
+#ifdef _LOG_NO_STD
+#undef _LOG_NO_STD
+#define _LOG_NO_STD 1
+#else
+#define _LOG_NO_STD 0
+#endif // _LOG_NO_STD
+
 #define LOG_MSG(LEVEL, f_str, ...)                                                   \
-    do {                                                                        \
-        fprintf(stderr, "[" STR(LEVEL) "]["  __FILE__  ":" STR(__LINE__) "] " f_str, ##__VA_ARGS__); \
+    do { \
+        if (!_LOG_NO_STD) {\
```

Comment:
This is not propper condition, the information that you will not log to the stdout or stderr is there in compile time.
Please use #if or #ifdef so it is resolved in compiletime instead of runtime